server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -113,6 +125,16 @@ public Builder fielddataFrequencyFilter(double minFreq, double maxFreq, int minS
             return builder;
         }
 
+        public Builder indexPrefixes(int minChars, int maxChars) {
+            if (minChars > maxChars)
+                throw new IllegalArgumentException("min_chars [" + minChars + "] must be less than max_chars [" + maxChars + "]");
+            if (minChars < 1)
+                throw new IllegalArgumentException("min_chars [" + minChars + "] must be greater than zero");
```

Comment:
not sure that we should let the user defines min and max or we should at least have a limit on maxChars ?
We could also have sane default values (`2` to `5` should be good enough for 99% of the case ?).

Reply:
> Maybe a max limit of something like 20

+1

> 2 to 5 sounds like a reasonable default though.

+1 too, changing these values should be considered as advanced (expert usage) ?